nano/secure/ledger.cpp

```diff
@@ -834,30 +831,18 @@ nano::uint128_t nano::ledger::balance (nano::block const & block)
 }
 
 // Balance for account containing hash
-nano::uint128_t nano::ledger::balance (store::transaction const & transaction, nano::block_hash const & hash) const
+std::optional<nano::uint128_t> nano::ledger::balance (store::transaction const & transaction, nano::block_hash const & hash) const
 {
 	if (hash.is_zero ())
 	{
 		return 0;
```

Comment:
Being consistent with the new way style of using optionals to indicate failed ops, should this perhaps return `std::nullopt` ?

Reply:
Made this change. It was a bit more tricky because ledger::amount was depending on this returning 0 if it couldn't find the hash.  I added an extra commit to change ledger::amount so it no longer depends on this behaviour.